static/js/settings_profile_fields.js

```diff
@@ -105,8 +106,23 @@ function update_choice_delete_btn($container, display_flag) {
     }
 }
 
+function get_value_for_new_option(container) {
+    const $choice_rows = $(container).find(".choice-row");
+    if ($choice_rows.length === 0) {
+        // Value for the first option is 0.
+        return 0;
+    }
+
+    const existing_option_values = [];
+    $choice_rows.each(function () {
+        existing_option_values.push(Number.parseInt($(this).attr("data-value"), 10));
+    });
+    existing_option_values.sort();
+    return existing_option_values[existing_option_values.length - 1] + 1;
```

Comment:
One other thing we can do here, rather than using the value greater than the already used largest value, we can instead use first number not used already starting from "0".

Reply:
We cannot do this since we want to handle the case where an existing option is deleted, then a new option added takes value of that deleted option and then we again add that option as mentioned [here](https://github.com/zulip/zulip/pull/22267/commits/d0d62530251fbf8b10c6488fda27346d41eb4975#r915077392) meaning same values for two options.